src/bin/download_sysext.rs

```diff
@@ -342,6 +391,59 @@ async fn main() -> Result<(), Box<dyn Error>> {
     fs::create_dir_all(&unverified_dir)?;
     fs::create_dir_all(&temp_dir)?;
 
+    // The default policy of reqwest Client supports max 10 attempts on HTTP redirect.
+    let client = Client::builder().redirect(Policy::default()).build()?;
+
+    // If input_xml exists, simply read it.
+    // If not, try to read from payload_url.
+    let res_local = match args.input_xml {
+        Some(name) => {
+            if name == "-".to_string() {
+                Some(io::read_to_string(io::stdin())?)
+            } else {
+                let file = File::open(name)?;
+                Some(io::read_to_string(file)?)
+            }
+        }
+        None => None,
+    };
+
+    let mut pkg_fake: Package;
+
+    let response_text = match res_local {
+        Some(res) => {
+            if args.payload_url.is_some() {
+                return Err(format!("Only one of the options can be given, --input-xml or --payload-url.").into());
```

Comment:
```suggestion
                return Err("Only one of the options can be given, --input-xml or --payload-url.".into());
```
Does this work? It's used in `test/crau_verify.rs`

Reply:
Removed the unnecessary format!.

Split into hash_on_disk_sha256 into a download library part.
Updated test/crau_verify to rely on the new lib function.